src/compiler/gcc.rs

```diff
@@ -279,30 +279,60 @@ fn compile<T>(creator: &T,
 {
     trace!("compile");
 
-    let output = match parsed_args.outputs.get("obj") {
+    let out_file = match parsed_args.outputs.get("obj") {
         Some(obj) => obj,
         None => {
             return future::err("Missing object file output".into()).boxed()
         }
     };
 
-    let mut cmd = creator.clone().new_command_sync(executable);
-    cmd.args(&["-c", "-x"])
-        .arg(match parsed_args.extension.as_ref() {
-            "c" => "cpp-output",
-            "cc" | "cpp" | "cxx" => "c++-cpp-output",
-            e => {
-                error!("gcc::compile: Got an unexpected file extension {}", e);
-                return future::err("Unexpected file extension".into()).boxed()
-            }
+    let extension = match parsed_args.extension.as_ref() {
```

Comment:
We should probably change `parsed_args.extension` into something like `enum SourceType { C, Cxx }`, shouldn't we? (That can be a followup.)